src/web3/WalletClient.ts

```diff
@@ -564,4 +570,28 @@ export class WalletClient extends BaseClient implements IWalletClient {
     }
     return [await sender.sellRolls(txData)];
   }
+
+  /**
+   * Validate an address.
+   *
+   * @param address - The address to validate.
+   *
+   * @returns A boolean (true if the address is valid, false otherwise).
+   */
+  validateAddress(address: string): boolean {
+    address = address.slice(ADDRESS_PREFIX.length); // remove The address prefix
+    // check if the address is base58 encoded
+    try {
+      // remove the version byte
+      const versionAndKeyBytes = new Uint8Array(base58Decode(address));
+      varintDecode(versionAndKeyBytes.slice(0, 1)); // check the version byte
```

Comment:
it seems that the varintDecode is done only on 1 byte. In Rust code, the version is encoded as a u64 varint so I guess its size could be more than 1 byte no?